src/bios.rs

```diff
@@ -0,0 +1,115 @@
+            meta
+        } else {
+            anyhow::bail!("No update metadata for component {} found", self.name());
+        };
+
+        let grub_install = Path::new("/").join(GRUB_BIN);
+        if !grub_install.exists() {
+            bail!("Failed to find {:?}", grub_install);
+        }
+
+        let boot_dir = Path::new(dest_root).join("boot");
+        
+        #[cfg(target_arch = "x86_64")]
+        let r = std::process::Command::new(grub_install)
+            .args(&["--target", "i386-pc"])
+            .args(&["--boot-directory", boot_dir.to_str().unwrap()])
+            .args(&["--modules", "mdraid1x"])
+            .arg(&device)
+            .status()?;
+        
+        #[cfg(target_arch = "powerpc64")]
+        let r = std::process::Command::new(grub_install)
+            .args(&["--target", "powerpc-ieee1275"])
+            .args(&["--boot-directory", boot_dir.to_str().unwrap()])
+            .arg("--no-nvram")
+            .arg(&device)
+            .status()?;
+
+        if !r.success() {
+            anyhow::bail!("Failed to run grub2-install");
+        }
+        Ok(InstalledContent {
+            meta,
+            filetree: None,
+            adopted_from: None,
+        })
+    }
+
+    fn generate_update_metadata(&self, sysroot_path: &str) -> Result<ContentMetadata> {
+        let grub_install = Path::new(sysroot_path).join(GRUB_BIN);
+        if !grub_install.exists() {
+            bail!("Failed to find {:?}", grub_install);
+        }
+        
+        // Query the rpm database and list the package and build times for all the
+        // files in the EFI system partition. If any files are not owned it is considered
+        // and error condition.
+        let rpmout = {
+            let mut c = ostreeutil::rpm_cmd(sysroot_path);
+            c.args(&["-q", "--queryformat", "%{nevra},%{buildtime} ", "-f"]);
+            c.arg(&grub_install);
+            c
+        }
+        .output()?;
+        if !rpmout.status.success() {
+            std::io::stderr().write_all(&rpmout.stderr)?;
+            bail!("Failed to invoke rpm -qf");
+        }        
+
+        let meta = util::parse_metadata(rpmout.stdout).unwrap();
+        write_update_metadata(sysroot_path, self, &meta)?;
+        Ok(meta)
+    }
+
+    fn query_adopt(&self) -> Result<Option<Adoptable>> {
+        unreachable!()
+    }
+
+    fn adopt_update(
+        &self,
+        sysroot: &openat::Dir,
+        update: &ContentMetadata,
+    ) -> Result<InstalledContent> {
+        unreachable!()
+    }
+
+    fn query_update(&self, sysroot: &openat::Dir) -> Result<Option<ContentMetadata>> {
+        get_component_update(sysroot, self)
+    }
+
+    fn run_update(
+        &self,
+        sysroot: &openat::Dir,
+        current: &InstalledContent,
+    ) -> Result<InstalledContent> {
+        unreachable!()
+    }
+
+    fn validate(&self, current: &InstalledContent) -> Result<ValidationResult> {
+        unreachable!()
```

Comment:
As discussed we could probably keep this unimplemented, though it should return an error or so?

Reply:
Return `Skipped` for `BIOS` instead, does that make sense?
```
[root@cosa-devsh ~]# bootupctl validate
Skipped: BIOS
Validated: EFI
```